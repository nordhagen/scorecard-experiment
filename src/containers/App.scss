@import '../global.scss';

.App {
  max-height: 100vh;
  background: black;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
}

.Header,
.TabBar {
  line-height: 0;
  img {
    width: 100%;
  }
}

.main {
  display: flex;
  flex-direction: row;
}

:global {
  .modal-enter {
    opacity: 0;
    transform: translate3d(10%, 0, 0);
  }
  .modal-enter-active {
    opacity: 1;
    transform: translateX(0);
    transition: opacity 300ms, transform 300ms;
  }
  .modal-exit {
    opacity: 1;
  }
  .modal-exit-active {
    opacity: 0;
    transform: translate3d(10%, 0, 0);
    transition: opacity 300ms, transform 300ms;
  }
  .modal-enter-done {
    width: 100%;
  }
}
