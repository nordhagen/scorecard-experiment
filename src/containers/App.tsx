import * as React from 'react'
import styles from './App.scss'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { clockTick, fetchEPG, closeModal } from '../state/actions'

import ButtonsView from './ButtonsView'
import ResultsView from './ResultsView'

class App extends React.Component {
  render() {
    return (
      <main className={styles.main}>
        <ButtonsView />
        <ResultsView />
      </main>
    )
  }
}

const mapStateToProps = state => ({
  nav: state.nav
})

const mapDispatchToProps = dispatch =>
  bindActionCreators({ clockTick, fetchEPG, closeModal }, dispatch)

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App)
