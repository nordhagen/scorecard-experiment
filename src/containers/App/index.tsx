import * as React from 'react'
import styles from './style.scss'

import ButtonsView from '../ButtonsView'
import ResultsView from '../ResultsView'

export default class App extends React.Component {
  render() {
    return (
      <main className={styles.main}>
        <ButtonsView />
        <ResultsView />
      </main>
    )
  }
}
