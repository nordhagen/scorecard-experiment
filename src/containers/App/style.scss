@import '/global.scss';

body,
html {
  min-height: 100vh;
}

body {
  margin: 0;
  font-family: sans-serif;
}

h1,
h2 {
  font-family: sans-serif;
  font-size: 2rem;
  font-weight: 100;
  color: white;
  margin: 0;
}

header {
  background-color: #18298c;
  padding: 20px;
}

.App {
  max-height: 100vh;
  background: black;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
}

.Header,
.TabBar {
  line-height: 0;
  img {
    width: 100%;
  }
}

.main {
  display: flex;
  flex-direction: row;
}

:global {
  .modal-enter {
    opacity: 0;
    transform: translate3d(10%, 0, 0);
  }
  .modal-enter-active {
    opacity: 1;
    transform: translateX(0);
    transition: opacity 300ms, transform 300ms;
  }
  .modal-exit {
    opacity: 1;
  }
  .modal-exit-active {
    opacity: 0;
    transform: translate3d(10%, 0, 0);
    transition: opacity 300ms, transform 300ms;
  }
  .modal-enter-done {
    width: 100%;
  }
}
