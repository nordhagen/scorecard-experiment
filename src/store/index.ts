import { createStore, combineReducers } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension'
import { scoreConfigReducer } from './config/reducers'
import { scoreStateReducer } from './score/reducers'

const rootReducer = combineReducers({
  score: scoreStateReducer,
  config: scoreConfigReducer
})

export type AppState = ReturnType<typeof rootReducer>
const store = createStore(rootReducer, composeWithDevTools())

export default store
