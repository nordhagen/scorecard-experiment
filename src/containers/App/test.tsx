import React from 'react'
import App from '.'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import store from '../../store'

test('It renders correctly', () => {
  const pastItem = renderer.create(
    <Provider store={store}>
      <App />)
    </Provider>
  )
  expect(pastItem.toJSON()).toMatchSnapshot()
})
