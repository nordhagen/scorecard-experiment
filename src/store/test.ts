import store from '.'

test('initial state', () => {
  let state
  state = store.getState()
  expect(state).toEqual({
    score: {
      items: [
        { type: 'a', count: 0 },
        { type: 'b', count: 0 },
        { type: 'c', count: 0 },
        { type: 'd', count: 0 }
      ]
    },
    config: {
      items: [
        { type: 'a', unitPoints: 50, bonusTrigger: 3, bonusValue: 200 },
        { type: 'b', unitPoints: 30, bonusTrigger: 2, bonusValue: 90 },
        { type: 'c', unitPoints: 20, bonusTrigger: null, bonusValue: null },
        { type: 'd', unitPoints: 15, bonusTrigger: null, bonusValue: null }
      ]
    }
  })
})
