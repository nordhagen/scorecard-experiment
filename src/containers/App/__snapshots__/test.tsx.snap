// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly 1`] = `
Array [
  <main>
    <section>
      <header>
        <h1>
          <span>
            Kahoot!
          </span>
           Points
        </h1>
      </header>
      <div>
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#F27781",
            }
          }
        >
          a
        </button>
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#04BF8A",
            }
          }
        >
          b
        </button>
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#F2CF1D",
            }
          }
        >
          c
        </button>
        <button
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#F29F05",
            }
          }
        >
          d
        </button>
      </div>
    </section>
    <section>
      <header>
        <h2>
          Player items
        </h2>
      </header>
      <div>
        <div>
          <table>
            <thead>
              <tr>
                <th>
                  Item
                </th>
                <th>
                  Qty
                </th>
                <th>
                  Score
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  a
                </td>
                <td>
                  0
                </td>
                <td>
                  0
                </td>
              </tr>
              <tr>
                <td>
                  b
                </td>
                <td>
                  0
                </td>
                <td>
                  0
                </td>
              </tr>
              <tr>
                <td>
                  c
                </td>
                <td>
                  0
                </td>
                <td>
                  0
                </td>
              </tr>
              <tr>
                <td>
                  d
                </td>
                <td>
                  0
                </td>
                <td>
                  0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div>
          <div>
            <p>
              Bonuses: 
              0
            </p>
          </div>
          <div>
            <h3>
              Total score:
            </h3>
            <p>
              0
            </p>
            <button
              onClick={[Function]}
              type="button"
            >
              New game
            </button>
          </div>
        </div>
      </div>
    </section>
  </main>,
  ")",
]
`;
